### PR TITLE
Filtering deployments by the '--release' argument

### PR DESCRIPTION
**Pull request by jsanemet**
On this pull request:
- Updating the 'get_deployment' query on a Zuul source so that it filters deployments by their release. Filtering is performed following a regex pattern.
- Adding the 'name' field to variants.

### CODE REVIEW

File: cibyl/plugins/openstack/sources/zuul/actions.py [hunk 1]
```diff
@@ -155,6 +113,105 @@ def get_target_jobs(self, **kwargs):
...
+            nodes={},
+            services={}
+        )
+
+
+class DeploymentFiltering:
+    """Takes care of applying the filters coming from the command line to a
+    deployment.
+    """
+    Filter = Callable[[Deployment], bool]
```
Comment (adrianfusco) [inline cibyl/plugins/openstack/sources/zuul/actions.py:164]: I don't get something. Shouldn't it be in the definition of a parameter or the return type of the method for example?

**Comment by bregman-arie** [comment]
Wanted to give it a try but right now I'm blocked on the bug mentioned in the chat

**Comment by cescgina** [comment]
Just tested it with `cibyl --release 16 --tenant tripleo-ci-internal --jobs periodic-tripleo-ci-rhel-8-scenario010-standalone-network-rhos-17` and got the following error:
```python
Traceback (most recent call last):
  File "/home/jgilaber/redhat/cibyl/.venv/cibyl/bin/cibyl", line 33, in <module>
    sys.exit(load_entry_point('Cibyl', 'console_scripts', 'cibyl')())
  File "/home/jgilaber/redhat/cibyl/cibyl/cli/main.py", line 118, in main
    orchestrator.query_and_publish(arguments["output_style"],
  File "/home/jgilaber/redhat/cibyl/cibyl/orchestrator.py", line 329, in query_and_publish
    self.run_query(system)
  File "/home/jgilaber/redhat/cibyl/cibyl/orchestrator.py", line 278, in run_query
    model_instances_dict = source_method(
  File "/home/jgilaber/redhat/cibyl/cibyl/plugins/openstack/sources/zuul/__init__.py", line 35, in get_deployment
    value=query.perform_query(**kwargs)
  File "/home/jgilaber/redhat/cibyl/cibyl/plugins/openstack/sources/zuul/actions.py", line 285, in perform_query
    if filters.is_valid_deployment(deployment):
  File "/home/jgilaber/redhat/cibyl/cibyl/plugins/openstack/sources/zuul/actions.py", line 209, in is_valid_deployment
    if not check(deployment):
  File "/home/jgilaber/redhat/cibyl/cibyl/plugins/openstack/sources/zuul/actions.py", line 194, in <lambda>
    lambda dpl: matches_regex(pattern, dpl.release.value)
  File "/home/jgilaber/redhat/cibyl/cibyl/utils/filtering.py", line 164, in matches_regex
    return bool(re.search(pattern, string))
  File "/usr/lib64/python3.10/re.py", line 200, in search
    return _compile(pattern, flags).search(string)
TypeError: expected string or bytes-like object
```

File: cibyl/plugins/openstack/sources/zuul/actions.py (hunk 1 above)
Comment (jsanemet) [reply cibyl/plugins/openstack/sources/zuul/actions.py:164]: Its below, on the constructor. The type of the 'filters' argument.

**Comment by jsanemet** [comment]
@cescgina You may try again now, I have covered the error.

**Pull request merged**